skglm/penalties/block_separable.py

```diff
@@ -330,3 +330,43 @@ def generalized_support(self, w):
                 gsupp[g] = True
 
         return gsupp
+
+
+class L1_1(BasePenalty):
+    """L1/1 row-wise penalty: sum of the coefficient magnitudes of the matrix."""
+
+    def __init__(self, alpha):
+        self.alpha = alpha
+
+    def get_spec(self):
+        spec = (
+            ('alpha', float64),
+        )
+        return spec
+
+    def params_to_dict(self):
+        return dict(alpha=self.alpha)
+
+    def value(self, W):
+        """Compute the L1/1 penalty value."""
+        return self.alpha * np.sum(np.abs(W))
+
+    def prox_1feat(self, value, stepsize, j):
+        """Compute proximal operator of the L2/1 penalty (block soft thresholding)."""
+        return ST_vec(value, self.alpha * stepsize)
+
+    def subdiff_distance(self, W, grad, ws):
+        """Compute distance of negative gradient to the subdifferential at W."""
+        subdiff_dist = np.zeros_like(ws, dtype=grad.dtype)
+        for idx, j in enumerate(ws):
+            if not np.any(W[j, :]):
+                # distance of - grad_j to alpha * the unit l\infty ball
+                norm_grad_j = np.max(np.abs(grad[idx, :]))
+                subdiff_dist[idx] = max(0, norm_grad_j - self.alpha)
+            else:
+                subdiff_dist[idx] = norm(grad[idx, :] + self.alpha * np.sign(W[j, :]))
```

Comment:
I've done the computations thrice, but I'm not able to find my mistake. I'd be happy to have feedbacks ;) @mathurinm

Reply:
I'm confused then. I thought :math`||A||_{1, 1} = \sum_{i=1}^n ||A_i||_1 = \sum_{i=1}^n \sum_{j=1}^p |A_{i,j}|`. As per the  first equality, I thought we could cast it in `skglm` BCD framework. So maybe, I should just loop over `subdiff_dist` and compute the 1-d subdiff_dist for every feature.

Reply:
Yes, but our working sets are defined row-wise (we perform BCD over the rows). Here since no structure is imposed, it's likely that there will often be at least one non zero coef per row, so working sets will consider the full matrix.

I am -1 on merging this because it would increase the size of the codebase. I believe this problem can be addressed by fitting several Lassos instead.

Reply:
@QB3 thoughts?